client/my-sites/plans/components/plans-header/index.tsx

```diff
@@ -64,17 +70,47 @@ const PlansHeader: React.FunctionComponent< {
 		subHeaderText ??
 		translate( 'See and compare the features available on each WordPress.com plan.' );
 
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const cart = useShoppingCart( selectedSiteId! );
+
 	if ( domainFromHomeUpsellFlow ) {
 		return <DomainUpsellHeader />;
 	}
 
+	async function handleAddCart() {
+		await cart.addProductsToCart( [
+			{
+				product_slug: PLAN_BUSINESS,
+				meta: 'agency-offer',
+			},
+			{
+				product_slug: PRODUCT_1GB_SPACE,
+				quantity: 50,
+				meta: 'agency-offer',
+			},
+		] );
+
+		window.location.href = `/checkout/${ selectedSiteSlug }`;
```

Comment:
I needed to change that as there was a race condition between the checkout URL and the `addProductsToCart` call. We need to wait for the products to be added, then redirect.

Reply:
I didn't experience a race condition here. I used the network tools to block the network call that mutated the cart and discovered that the navigation doesn't happen until/unless the cart edits have finished.

i.e. the `await` appears to be doing it's job fine to me.